controllers/clusterversion_controller.go

```diff
@@ -124,47 +122,181 @@ func (c *ClusterVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
+		c.log.Error(err, "failed to get the deployment for the console")
+		return ctrl.Result{}, err
+	}
+
+	// Load the operator configmap
+	c.operatorConfigMap.Name = operatorConfigMapName
+	c.operatorConfigMap.Namespace = c.OperatorNamespace
+	if err := c.get(&c.operatorConfigMap); err != nil && !k8serrors.IsNotFound(err) {
 		return ctrl.Result{}, err
 	}
 
-	instance := configv1.ClusterVersion{}
-	if err = c.Client.Get(context.TODO(), req.NamespacedName, &instance); err != nil {
+	c.log.Info("Starting reconciliation")
+	if result, err := c.reconcilPhases(); err == nil {
+		c.log.Error(err, "An error was encountered during reconciliation")
 		return ctrl.Result{}, err
+	} else {
+		c.log.Info("Reconciliation completed successfully")
+		return result, nil
 	}
+}
+
+func (c *ClusterVersionReconciler) reconcilPhases() (ctrl.Result, error) {
 
-	if err := csi.InitializeSidecars(c.log, instance.Status.Desired.Version); err != nil {
+	// Reconcile console related resources
+	if err := c.reconcileNginxConfigMap(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := c.reconcileConsoleService(); err != nil {
+		return ctrl.Result{}, err
+	}
+	if err := c.reconcileConsolePlugin(); err != nil {
+		c.log.Error(err, "unable to deploy client console")
+		return ctrl.Result{}, err
+	}
+
+	// Reconcile CSI related resources
+	if err := csi.InitializeSidecars(c.log, c.clusterVersion.Status.Desired.Version); err != nil {
 		c.log.Error(err, "unable to initialize sidecars")
 		return ctrl.Result{}, err
 	}
+	if err := c.reconcileSecurityContextConstraints(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileMonConfigMap(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileEncConfigMap(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileCephFSDeployment(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileCephFSDaemonSet(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileRBDDeployment(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileRBDDaemonSet(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileCephFSCSIDrivers(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcileRBDCSIDriver(); err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := c.reconcilePrometheusRule(); err != nil {
+		return reconcile.Result{}, err
+	}
 
-	c.scc = &secv1.SecurityContextConstraints{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: csi.SCCName,
-		},
+	return reconcile.Result{}, nil
+}
+
+func (c *ClusterVersionReconciler) reconcileNginxConfigMap() error {
+	c.log.Info("Reconciling NginxConfigMap")
+
+	nginxConfigMap := corev1.ConfigMap{}
+	nginxConfigMap.Name = console.NginxConfigMapName
+	nginxConfigMap.Namespace = c.OperatorNamespace
+
+	err := c.createOrUpdate(&nginxConfigMap, func() error {
+		if err := c.own(&nginxConfigMap); err != nil {
```

Comment:
in existing code, console deployment owns the configmap and that seems to be correct and so setting another controller ref here always returns an error.

Reply:
Console deployment should not own the config map because the deployment controller is not the one that reconciles it. But I guess you mean that the execution fails because there is an existing controller ref.

Reply:
> you mean that the execution fails because there is an existing controller ref.
- yes.